Testinvi/Tweetinvi/AsyncAndMultiThreading/AsyncCredentialsTests.cs

```diff
@@ -0,0 +1,267 @@
+﻿using System;
+using System.Configuration;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Tweetinvi;
+using Tweetinvi.Models;
+
+namespace Testinvi.Tweetinvi.AsyncAndMultiThreading
+{
+    [TestClass]
+    public class AsyncCredentialsTests
+    {
+        #region Async/Await
+
+        [TestMethod]
+        public async Task AsyncLocalCredsAreCarriedOutToChild()
+        {
+            // Arrange
+            var creds = new TwitterCredentials("a", "a", "a", "a");
+            Auth.SetCredentials(creds);
+
+            // Act
+            await Sync.ExecuteTaskAsync(() =>
+            {
+                // Assert
+                Assert.AreNotEqual(Auth.Credentials, creds);
+                AssertAreCredentialsEquals(Auth.Credentials, creds);
+            });
+        }
+
+        [TestMethod]
+        public async Task AsyncLocalCredsAreNotCarriedOutToParent()
+        {
+            await AsyncLocalCredsAreNotCarriedOutToParentRecursively(0);
+        }
+
+        [TestMethod]
+        public async Task AsyncLocalCredsAreNotCarriedOutToParentMultipleLevels()
+        {
+            await AsyncLocalCredsAreNotCarriedOutToParentRecursively(5);
+        }
+
+
+        public async Task AsyncLocalCredsAreNotCarriedOutToParentRecursively(int i)
```

Comment:
At some point I want to migrate to another Framework but I have not used any other yet and I need to find out what are the benefits of those.

As for your comment, what do you mean, there is `Theory` equivalent?

Reply:
I probably shouldn't have used the word theory. It's just what xUnit calls a parametrised test.
So here instead of multiple test methods calling this one we'd have something like:
```cs
[Theory]
[InlineData(0)]
[InlineData(5)]
public async Task AsyncLocalCredsAreNotCarriedOutToParentRecursively(int i)
...
```